reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/TakeObservable.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+/**
+ * Emit only the first [limit] items emitted by source.
+ */
+fun <T> Observable<T>.take(limit: Int): Observable<T?> {
+    require(limit > 0) { "count >= 0 required but it was $limit" }
+
+    return observable { emitter ->
+        val disposableWrapper = DisposableWrapper()
+        emitter.setDisposable(disposableWrapper)
+
+        var remaining = limit
+
+        subscribeSafe(object : ObservableObserver<T?> {
+            override fun onComplete() {
+                emitter.onComplete()
+            }
+
+            override fun onError(error: Throwable) {
+                emitter.onError(error)
+            }
+
+            override fun onNext(value: T?) {
+                if (remaining-- > 0) {
```

Comment:
@arkivanov As I remember we don't need any synchronisation here, right?

Reply:
All `Observer`'s callbacks are never executed concurrently, but may be executed from different threads. As we are using atomics then no syncrhonization is needed.